crates/rust-analyzer/src/lsp/utils.rs

```diff
@@ -75,25 +75,21 @@ impl GlobalState {
     }
 
     /// Sends a notification to the client containing the error `message`.
-    /// If `additional_info` is [`Some`], appends a note to the notification telling to check the logs.
-    /// This will always log `message` + `additional_info` to the server's error log.
+    /// If `additional_info` is [`Some`], the details will be included in the server's logs.
     pub(crate) fn show_and_log_error(&mut self, message: String, additional_info: Option<String>) {
         match additional_info {
             Some(additional_info) => {
                 tracing::error!("{}:\n{}", &message, &additional_info);
-                self.show_message(
-                    lsp_types::MessageType::ERROR,
-                    message,
-                    tracing::enabled!(tracing::Level::ERROR),
-                );
             }
             None => {
                 tracing::error!("{}", &message);
-                self.send_notification::<lsp_types::notification::ShowMessage>(
-                    lsp_types::ShowMessageParams { typ: lsp_types::MessageType::ERROR, message },
-                );
             }
         }
+        self.show_message(
+            lsp_types::MessageType::ERROR,
+            message,
+            tracing::enabled!(tracing::Level::ERROR),
+        );
```

Comment:
This unnecessarily opens the log button notification now when there is no additional info in the logs

Reply:
where should error messages from flycheck be printed? flycheck eagerly gets that, but printing with additional information interrupts the error message.

that's probably a sign that the the stuff in flycheck isn't sufficient here, huh

Reply:
I don't think I understand, why is flycheck relevant here? This is just for logging errors, if the error has no extra information we show all the info in the popup, if it has additional info we log that and tell the user to check the logs by giving the pop up a button. I'm a bit confused as to why you need to make changes here.

Reply:
right, sorry: it's the interaction of several different features. 
-  `tracing::enabled!(tracing::Level::ERROR)` is almost always true (since it's the default level for `tracing-subscriber`)
- `CargoActor` in `flycheck` will be printing out the lines it couldn't parse as JSON (which will often contain the error itself...), but that's interleaved with the `additional_info`.

If I'm *not* make this change, the end result is the following (where the `additional_info` is "unable to do the things"): 

<details>
<summary>Screenshot of mangled logs</summary>
<img width="1357" alt="Screenshot 2024-05-23 at 11 18 14 AM" src="https://github.com/rust-lang/rust-analyzer/assets/2067774/b3730b82-2384-4983-b5af-1a91ce23d2ff">
</details>

> that's probably a sign that the the stuff in flycheck isn't sufficient here, huh

This is what I meant when I said "that's probably a sign that the the stuff in flycheck isn't sufficient here, huh". However, having thought about it, it's not that flycheck is incorrect! It's that the CLI that i'm calling here, rust-project](https://github.com/facebook/buck2/tree/main/integrations/rust-project), that's wrong. It should communicate the underlying error in a more structured matter.

I'll hack on something today and propose something in this PR, which would let me undo the change here.